### ISSUE TITLE
moco-controller keeps restarting

### ISSUE THREAD

**Issue by uqix** [cybozu-go/moco#463]
**Describe the bug**
moco-controller keeps restarting.

**Environments**
- Version: helm 0.3.0
- OS: k8s v1.20.12, centos 7.8

**Additional context**
* moco-controllers run on nodes without internet connection

Error logs:

```
{"level":"error","ts":1664248990.236583,"logger":"controller-runtime.source","msg":"if kind is a CRD, it should be installed before calling Start","kind":"PodDisruptionBudget.policy","error":"no matches for kind \"PodDisruptionBudget\" in version \"policy/v1\"","stacktrace":"sigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start.func1.1\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/source/source.go:139\nk8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext\n\t/go/pkg/mod/k8s.io/apimachinery@v0.24.3/pkg/util/wait/wait.go:233\nk8s.io/apimachinery/pkg/util/wait.WaitForWithContext\n\t/go/pkg/mod/k8s.io/apimachinery@v0.24.3/pkg/util/wait/wait.go:660\nk8s.io/apimachinery/pkg/util/wait.poll\n\t/go/pkg/mod/k8s.io/apimachinery@v0.24.3/pkg/util/wait/wait.go:594\nk8s.io/apimachinery/pkg/util/wait.PollImmediateUntilWithContext\n\t/go/pkg/mod/k8s.io/apimachinery@v0.24.3/pkg/util/wait/wait.go:545\nsigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start.func1\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/source/source.go:132"}
...
Error: failed to wait for mysqlcluster caches to sync: timed out waiting for cache to be synced
```

Full logs:

see attachment: [moco-controller-7bbd5bb6f9-8246z.log](https://github.com/cybozu-go/moco/files/9651260/moco-controller-7bbd5bb6f9-8246z.log).

**Comment by uqix** [comment]
Downgrading to helm 0.2.3 works

**Comment by ymmt2005** [comment]
@d-kuro could you take a look?

**Comment by d-kuro** [comment]
@uqix 

Chart version v0.3.0 uses MOCO v0.13.0 by default.[^1]

[^1]:https://github.com/cybozu-go/moco/blob/gh-pages/index.yaml

MOCO v0.13.0 supports three Kubernetes versions 1.22, 1.23, and 1.24.[^2]
The `batch/v1/cronjobs` and `policy/v1/poddisruptionbudgets` APIs used by MOCO are not available in Kubernetes v1.20.
I believe this is why the error is occurring.

[^2]:https://github.com/cybozu-go/moco/tree/v0.13.0#supported-software

You can upgrade your Kubernetes to a supported version or specify an older version of the MOCO tag in the Helm chart values.[^3]

[^3]:https://github.com/cybozu-go/moco/blob/v0.13.0/charts/moco/values.yaml#L7

**Comment by uqix** [comment]
Somehow I missed the doc, anyway maybe we can add some helm hooks to validate the k8s version?
